features/admin.connections.v1/components/edit/settings/custom-authenticator-settings.tsx

```diff
@@ -0,0 +1,529 @@
+     */
+    onUpdate: (id: string, tabName?: string) => void;
+}
+
+/**
+ * Custom authenticator settings page.
+ *
+ * @param props - Props injected to the component.
+ * @returns React element.
+ */
+export const CustomAuthenticatorSettings: FunctionComponent<CustomAuthenticatorSettingsPagePropsInterface> = ({
+    isCustomLocalAuthenticator,
+    isLoading,
+    connector,
+    onUpdate,
+    ["data-componentid"]: componentId = "authenticator-settings-page"
+}: CustomAuthenticatorSettingsPagePropsInterface): ReactElement => {
+    const dispatch: Dispatch = useDispatch();
+
+    const [ authenticatorEndpoint, setAuthenticatorEndpoint ] = useState<EndpointConfigFormPropertyInterface>(
+        null
+    );
+    const [ endpointAuthenticationType, setEndpointAuthenticationType ] = useState<AuthenticationType>(null);
+    const [ isAuthenticationUpdateFormState, setIsAuthenticationUpdateFormState ] = useState<boolean>(false);
+
+    const { t } = useTranslation();
+
+    /**
+    * This useEffect is utilized only for custom authenticators in order to fetch additional
+    * details related to authenticators.
+    * This is not required for other connections since all the required details
+    * are passed from the parent component.
+    */
+    useEffect(() => {
+
+        let customAuthenticatorId: string;
+
+        if (isCustomLocalAuthenticator) {
+            customAuthenticatorId = (connector as CustomAuthConnectionInterface)?.id;
+            getCustomLocalAuthenticator(customAuthenticatorId);
+        } else {
+            customAuthenticatorId = connector?.federatedAuthenticators?.authenticators[0]?.authenticatorId;
+            getCustomFederatedAuthenticator(customAuthenticatorId);
+        }
+    }, [ isCustomLocalAuthenticator ]);
+
+    const resolveDisplayName = (): string => {
+        if (isCustomLocalAuthenticator) {
+            return (connector as CustomAuthConnectionInterface)?.displayName;
+        } else {
+            return (connector as ConnectionInterface)?.name;
+        }
+    };
+
+    /**
+    * This function is used to get the custom local authenticator details which includes endpoint
+    * configurations that need to be accessed from the "Settings" tab.
+    *
+    * @param customLocalAuthenticatorId - Custom local authenticator id.
+    */
+    const getCustomLocalAuthenticator = (customLocalAuthenticatorId: string) => {
+        getLocalAuthenticator(customLocalAuthenticatorId)
+            .then((data: CustomAuthConnectionInterface) => {
+                const endpointAuth: EndpointConfigFormPropertyInterface = {
+                    authenticationType: data?.endpoint?.authentication?.type,
+                    endpointUri: data?.endpoint?.uri
+                };
+
+                setAuthenticatorEndpoint(endpointAuth);
+            })
+            .catch((error: IdentityAppsApiException) => {
+                if (error.response && error.response.data && error.response.data.description) {
```

Comment:
Let's use optional chaining for better readability.

https://github.com/wso2/identity-apps/blob/master/docs/write-code/CODE_QUALITY.md#use-modern-js-features-to-keep-the-code-concise-and-readable

```suggestion
                if (error?.response?.data?.description) {
```

Reply:
This is updated with the commit 0bb86d3e341eff042933dbc1aeb8eef2a2b4b383